docs/book/developer-guide/repo-and-config.md

```diff
@@ -2,8 +2,309 @@
+
+{% endhint %}
+
+The first time that ZenML is run on a machine, it creates the _Global Config
+Directory_ and initializes the default configuration in it, along with a default
+profile and stack:
+
+```
+$ zenml stack list
+Unable to find ZenML repository in your current working directory (/home/stefan)
+or any parent directories. If you want to use an existing repository which is in
+a different location, set the environment variable 'ZENML_REPOSITORY_PATH'. If
+you want to create a new repository, run zenml init.
+Initializing the ZenML global configuration version to 0.7.3
+Creating default profile...
+Initializing profile default...
+Registering default stack...
+Registered stack component with type 'orchestrator' and name 'default'.
+Registered stack component with type 'metadata_store' and name 'default'.
+Registered stack component with type 'artifact_store' and name 'default'.
+Registered stack with name 'default'.
+Created and activated default profile.
+Running without an active repository root.
+Running with active profile: 'default' (global)
+┏━━━━━━━━┯━━━━━━━━━━━━┯━━━━━━━━━━━━━━━━┯━━━━━━━━━━━━━━━━┯━━━━━━━━━━━━━━┓
+┃ ACTIVE │ STACK NAME │ ARTIFACT_STORE │ METADATA_STORE │ ORCHESTRATOR ┃
+┠────────┼────────────┼────────────────┼────────────────┼──────────────┨
+┃   👉   │ default    │ default        │ default        │ default      ┃
+┗━━━━━━━━┷━━━━━━━━━━━━┷━━━━━━━━━━━━━━━━┷━━━━━━━━━━━━━━━━┷━━━━━━━━━━━━━━┛
+```
+
+The following is an example of the layout of the _Global Config Directory_
+immediately after initialization:
+
+```
+/home/stefan/.config/zenml   <- Global Config Directory
+├── config.yaml              <- Global Configuration Settings
+├── local_stores             <- Every stack component that stores information
+|   |                        locally will have its own subdirectory here.
+|   |                        
+│   └── 09fcdb1c-4079-4d20-afdb-957965405863   <- Local Store path for the `default`
+|                                              local Artifact Store
+|
+└── profiles                 <- root path where Profiles data (stacks, components,
+    |                        etc) are stored by default. Every profile will have
+    |                        its own subdirectory here, unless the profile is
+    |                        configured with a custom configuration path.
+    |
+    └── default              <- configuration folder for the `default` profile.
+        ├── artifact_stores
+        │   └── default.yaml
+        ├── metadata_stores
+        │   └── default.yaml
+        ├── orchestrators
+        │   └── default.yaml
+        └── stacks.yaml
+```
+
+As shown above, the _Global Config Directory_ stores the following
+information:
+
+1. The `global.yaml` file stores the global configuration settings: the unique
+ZenML user ID, the active Profile, the analytics related options and a list of all configured Profiles, along with their configuration attributes, such as the
```

Comment:
Just noting that we seem to sometimes capitalise 'Profile' and sometimes not. Maybe pick one for the sake of consistency?